pkg/hash/hash.go

```diff
@@ -78,19 +67,22 @@ func (hash *Hash) WriteAny(data ...interface{}) (int64, error) {
 	for _, d := range data {
 		switch t := d.(type) {
 		case []byte:
-			n0, _ := hash.h.Write(t)
+			n0, err := hash.Write(t)
+			if err != nil {
+				return n, fmt.Errorf("hash.Hash: write []byte: %w", err)
+			}
 			n += int64(n0)
 		case []*curve.Point:
 			for _, p := range t {
-				n0, err := p.WriteTo(hash.h)
+				n0, err := p.WriteTo(hash)
 				n += n0
 				if err != nil {
 					return n, fmt.Errorf("hash.Hash: write []*curve.Point: %w", err)
 				}
 			}
 		case []curve.Point:
 			for _, p := range t {
-				n0, err := p.WriteTo(hash.h)
+				n0, err := p.WriteTo(hash)
 				n += n0
 				if err != nil {
 					return n, fmt.Errorf("hash.Hash: write []curve.Point: %w", err)
```

Comment:
These two cases are no longer necessary since a few commits ago, I think we can remove them (and remove the test for this in `hash_test.go`)

Reply:
Sounds good